data/rulesets.js

```diff
@@ -549,7 +549,7 @@ exports.BattleFormats = {
 					}
 				}
 			}
-			if (!nonSpeedBoosted) return;
+			if (!nonSpeedBoosted || (speedBoosted !== nonSpeedBoosted && typeof speedBoosted === 'string' && typeof nonSpeedBoosted === 'string')) return;
```

Comment:
This logic is kind of convoluted.

Should at least be split into two `if`s.

Then the second one is

`if (speedBoosted !== nonSpeedBoosted && typeof speedBoosted === 'string' && typeof nonSpeedBoosted === 'string') return;`